prow/hook/metrics.go

```diff
@@ -0,0 +1,27 @@
+package hook
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// Define all metrics for webhooks here.
+	webhookCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "prow_webhook_counter",
+		Help: "A counter of the webhooks made to prow.",
+	}, []string{"event_type"})
+)
+
+func init() {
+	prometheus.MustRegister(webhookCounter)
+}
+
+type Metrics struct {
+	WebhookCounter *prometheus.CounterVec
```

Comment:
I guess we'll also want individual plugins to make their own metrics. Should those live in this file, or should they be in the actual plugin packages?

Reply:
If we ever want to do actual go plugins, I guess each plugin should contain its own metrics. Another thing to think about in that case is that we need to push metrics from plugins as opposed to having Prometheus pull them?

Reply:
maybe each plugin want to register its metrics through hook?

Reply:
https://prometheus.io/docs/instrumenting/pushing/

Reply:
pushing looks easier, do we already have a pushgateway available? <- @apelisse

Reply:
Push metrics would be preferable, because we're running multiple instances, and the only way to hit them is through an LB.

Reply:
> pushing looks easier, do we already have a pushgateway available?

We have an influx database that we can push to. I think @cjwagner already does that.

Reply:
I've used both influx and prometheus to display data on Velodrome, but I haven't pushed to prometheus before. I think its probably easier to set up a prometheus pushgateway than it is to use both prometheus and influxdb. You could use influxdb for everything, but prometheus polling is simpler for things like webhook counters.

Reply:
@cjwagner we have multiple instances of prow, and they're only visible behind a load balancer. I think we'll need to push them out.

Reply:
> Polling is tricky because hook is replicated in a separate cluster behind a load balancer. It seems like pushing is the way to go.

@smarterclayton what will Prometheus do if I have annotated a service with multiple pods for scraping? Will it aggregate the metrics?